pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/GuardLogStatementRule.java

```diff
@@ -171,23 +185,54 @@ private String getLastPartOfName(String name) {
         return result;
     }
 
+    /**
+     * Gets the first child, first grand child, ... of the given types.
+     * The children must follow the given order of types
+     *
+     * @param root the node from where to start the search
+     * @param childrenTypes the list of types
+     * @param <N> should match the last type of childrenType, otherwise you'll get a ClassCastException
+     * @return the found child node or <code>null</code>
+     */
+    @SafeVarargs
+    private static <N> N getFirstChild(Node root, Class<? extends Node> ... childrenTypes) {
```

Comment:
this is interesting, something we may want to even move to `AbstractNode` maybe?

Reply:
Also, it should probably be `<N extends Node> N`